src/Microsoft.ServiceFabric.Actors.Migration/KVStoRCMigrationActorStateProvider.cs

```diff
@@ -356,6 +358,149 @@ internal IReliableStateManagerReplica2 GetStateManager()
             return this.rcStateProvider.GetStateManager();
         }
 
+        internal async Task<MigrationState> GetMigrationStateAsync()
+        {
+            var metaDataDictionary = await this.GetMetadataDictionaryAsync();
+            ConditionalValue<byte[]> migrationStateValue;
+            using (var tx = this.GetStateManager().CreateTransaction())
+            {
+                migrationStateValue = await metaDataDictionary.TryGetValueAsync(tx, MigrationConstants.MigrationStateKey);
+            }
+
+            MigrationState.TryParse(Encoding.ASCII.GetString(migrationStateValue.Value), out MigrationState migrationState);
+
+            return migrationState;
+        }
+
+        internal async Task<MigrationStatus> GetMigrationStatusAsync(CancellationToken cancellationToken)
+        {
+            var metaDataDictionary = await this.GetMetadataDictionaryAsync();
+            var migrationStatus = new MigrationStatus();
+            var workerStatuses = new List<WorkerStatus>();
+
+            migrationStatus.ParitionId = this.servicePartition.PartitionInfo.Id;
+
+            ConditionalValue<byte[]> migrationPhaseValue;
+            cancellationToken.ThrowIfCancellationRequested();
+            using (var tx = this.GetStateManager().CreateTransaction())
+            {
+                migrationPhaseValue = await metaDataDictionary.TryGetValueAsync(tx, MigrationConstants.MigrationPhaseKey);
+            }
+
+            if (migrationPhaseValue.HasValue)
+            {
+                MigrationPhase.TryParse(Encoding.ASCII.GetString(migrationPhaseValue.Value), out MigrationPhase phase);
+                migrationStatus.CurrentMigrationPhase = phase.ToString();
+
+                ConditionalValue<byte[]> migrationStartTimeUtcValue, currentMigrationPhaseStartTimeUtcValue;
+                ConditionalValue<byte[]> lastAppliedSNValue;
+                long lastAppliedSN, startSN, lastSN;
+
+                cancellationToken.ThrowIfCancellationRequested();
+                using (var tx = this.GetStateManager().CreateTransaction())
+                {
+                    migrationStartTimeUtcValue = await metaDataDictionary.TryGetValueAsync(tx, MigrationConstants.MigrationStartTimeUtcKey);
+                    currentMigrationPhaseStartTimeUtcValue = await metaDataDictionary.TryGetValueAsync(tx, MigrationConstants.CurrentMigrationPhaseStartTimeUtcKey);
+                }
+
+                DateTime.TryParse(Encoding.ASCII.GetString(migrationStartTimeUtcValue.Value), out DateTime migrationStartTimeUtc);
```

Comment:
Either proceed based on return value of TryParse or use simply use Parse

Reply:
Done